streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -293,6 +328,8 @@
      *                          if not specified the default serdes defined in the configs will be used
      * @param <V1>              the value type of the other stream
      * @param <R>               the value type of the new stream
+     *
+     * @return a {@link KStream} containing the join-tuples computed by the given {@link ValueJoiner}
```

Comment:
Nitpick: Is there a better term than "join-tuples"?  Perhaps we should highlight a bit more that that we return tuples of values (but e.g. not of keys).